spec/formatting.md

```diff
@@ -18,3 +18,71 @@ the local variable takes precedence.
 
 It is an error for a local variable definition to
 refer to a local variable that's defined after it in the message.
+
+## Error Handling
+
+During the formatting of a message,
+various errors may be encountered.
+These are divided into the following categories:
+
+- **Syntax errors** occur when the syntax representation of a message is invalid.
+- **Resolution errors** occur when the runtime value of a part of a message
+  cannot be determined.
+
```

Comment:
One type of error that I think might be missing in this categorization: validity of the options.
For example `{$foo :dateformat mihai=13 length=pink}`
Where the key `mihai` is invalid, and key `length` is valid, but the value (`pink`) is not.

Reply:
I would think that this would be a Formatting Error, yes? I don't think it's reasonable for us to expect or require the function registry to be absolutely exhaustive in e.g. enumerating all the options baskets that are supported, so we would only be able to discover this failure at the time when we've acquired the value references by `$foo` and are trying to format it.

Reply:
I'd also consider it to be a type hinting which can help CAT tools to warn, but shouldn't be required for the registry entry to work.
And if such error happens at runtime I'd expect the `mihai` to be ignored (sorry Mihai! ;)) and `length=pink` to be ignored too.

Reply:
@zbraniecki  mentioned:
> And if such error happens at runtime I'd expect the mihai to be ignored (sorry Mihai! ;)) and length=pink to be ignored too.

I would expect an unrecognized option (`mihai` in this case) to be ignored. I would expect that it would be implementation dependent if a recognized option (`length` in the example) had an invalid value. For example, I could see an implementation wanting to throw what amounts to `IllegalArgumentException` in some cases or just use defaulting behavior in others.

On an abstract level our spec can define *when* errors ***may*** be thrown (or, in some cases, when they ***must*** be thrown) and *what type* of error to expect when one is thrown.

Reply:
I'm fine with formatting error, even if I am ignored ;-)